prov/gni/test/rdm_atomic.c

```diff
@@ -481,7 +484,7 @@ void do_min(int len)
 
 	cr_assert_eq(ret, 1, "fi_cq_read returned %d %d", ret,
 			dump_cq_error(send_cq[0], target, 0));
-	rdm_atomic_check_tcqe(&cqe, target, FI_ATOMIC | FI_WRITE, 0);
+	rdm_atomic_check_tcqe(&cqe, target, FI_ATOMIC | FI_WRITE, 0, ep[1]);
```

Comment:
This check makes no sense.  You are checking the data field on the initiating (send) side.  In order to test for the remote cq data, you need to request for remote CQ events on ep[1] and look at the events delivered there.  This is true of all the checks you added in this file.